### ISSUE TITLE
[Support]: Odd behaviour when deferring packages installed through Nix (PDF-Tools)

### ISSUE THREAD

**Issue by Icy-Thought** [progfolio/elpaca#185]
### Elpaca Version

Elpaca 9a130c0 HEAD -> master, origin/master, origin/HEAD
installer:      0.5
emacs-version:  GNU Emacs 29.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.16.0)

### Operating System

NixOS (unstable)

### Description

I have encountered two instances of Emacs packages installed through the Nix package manager that displayed some form of odd behaviour when supplied with both `:defer t` and `:elpaca nil`. (Binaries provided by Nix)

```emacs-lisp
(use-package pdf-tools
  :elpaca nil ;; <-^ fetched from Nixpkgs
  :config (push 'pdf-tools elpaca-ignored-dependencies))

(use-package pdf-view
  :elpaca nil
  :after pdf-tools
  :magic ("%PDF" . pdf-view-mode)
  :mode ("\\.[pP][dD][fF]\\'" . pdf-view-mode)
  :hook (pdf-view-mode . (lambda ()
                           ;; :NOTE| Set the PDF free from the unnecessary borders
                           (set (make-local-variable 'evil-normal-state-cursor) (list nil))
                           (pdf-view-themed-minor-mode)))
  :config
  (define-advice pdf-view-enlarge (:after (&rest _args) center-after-enlarge)
    "Center the PDF view in the active PDF window after enlarging it."
   (pdf-view-center-in-window))

  (define-advice pdf-view-shrink (:after (&rest _args) center-after-shrink)
    "Center the PDF view in the active PDF window after shrinking it."
    (pdf-view-center-in-window))
  :custom
  (pdf-view-use-scaling t)
  (pdf-view-use-imagemagick nil)
  (pdf-view-display-size 'fit-width))

(use-package pdf-view-restore
  :hook (pdf-view-mode . pdf-view-restore-mode)
  :custom (pdf-view-restore-filename (no-littering-expand-var-file-name "pdf-view-restore")))
```

The recent encounter of this odd behaviour occured when I attempted to defer [PDF-Tools](https://github.com/vedang/pdf-tools).

Where I had noticed that the value of `pdf-info-epdfinfo-program` had changed to:

```
"/home/chaoticmist/.config/emacs/var/elpaca/builds/pdf-tools/epdfinfo"
```

from the non-deferred value:

```
"/nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-packages-deps/share/emacs/site-lisp/elpa/pdf-tools-20230611.239/epdfinfo"
```

I have tried verifying the presence of `/nix/store` related path's in `list-load-path-shadows` and surely enough they were present. Among them was:

```
/home/chaoticmist/.config/emacs/var/elpaca/builds/pdf-tools/pdf-tools hides /nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-packages-deps/share/emacs/site-lisp/elpa/pdf-tools-20230611.239/pdf-tools
```

and when I had execute `ls` in the `/nix/store` directory path of `PDF-Tools`, the `epdfinfo` binary file was present among the files listed within that directory. Something that I find particularly odd though is the lack of `epdfinfo` in the `list-load-path-shadows`, whereas other files are being detected as expected.

List of files being detected + listed in `list-load-path-shadows`:
```
pdf-macs
pdf-sync
pdf-virtual
pdf-history
pdf-view
pdf-isearch
pdf-outline
pdf-misc
pdf-util
pdf-tools
pdf-cache
pdf-loader
pdf-dev
pdf-tools-autoloads
pdf-links
pdf-info
pdf-annot
pdf-occur
```

**Comment by progfolio** [comment]
Try adding this as early as possible in your init file.

``` emacs-lisp
(add-variable-watcher 'load-path
 (lambda (sym new op where)
   (when (cl-some (lambda (path) (string-match-p "elpaca/builds/pdf-tools" path)) new)
     (debug))))
```

It should hit the debugger when elpaca's build dir for pdf-tools is added to the load-path.
Copy the entire expanded backtrace here.

**Comment by Icy-Thought** [comment]
```emacs-lisp
Debugger entered: nil
  (progn (debug))
  (if (cl-some #'(lambda (path) (string-match-p "elpaca/builds/pdf-tools" path)) new) (progn (debug)))
  (closure (t) (sym new op where) (if (cl-some #'(lambda (path) (string-match-p "elpaca/builds/pdf-tools" path)) new) (progn (debug))))(load-path ("/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." "/home/icy-thought/Workspace/public/emacs.d/irkalla" "/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-..." "/nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-..." "/nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-..." "/nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-..." "/nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-..." "/nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-..." "/nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-..." "/nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-..." "/nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-..." "/nix/store/q4hhfyqb40mahq5d77b7wpl8ccr4ggzp-emacs-..." ...) set nil)
  elpaca--activate-package((elpaca pdf-tools "pdf-tools" pdf-tools (activation info unblocked blocked queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/p..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "pdf-tools" :fetcher github :repo "vedang/pdf-tools" :files (:defaults "README" ("build" "Makefile") ("build" "server")) :protocol https :inherit t :depth 1) nil (tablist) (pdf-view-restore) ((emacs "26.3") (tablist "1.0") (let-alist "1.0.4")) 2 (25870 57704 765632 925000) t nil ((activation (25870 57704 767026 971000) "Activating package" 0) (info (25870 57704 767021 803000) "No Info dir file found" 0) (unblocked (25870 57704 766989 326000) "unblocked by dependency tablist" 0) (queueing-deps (25870 57704 765652 341000) "Queueing Dependencies" 1) (queued (25870 57704 765638 163000) "Package queued" 1)) t))
  elpaca--continue-build((elpaca pdf-tools "pdf-tools" pdf-tools (activation info unblocked blocked queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/p..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "pdf-tools" :fetcher github :repo "vedang/pdf-tools" :files (:defaults "README" ("build" "Makefile") ("build" "server")) :protocol https :inherit t :depth 1) nil (tablist) (pdf-view-restore) ((emacs "26.3") (tablist "1.0") (let-alist "1.0.4")) 2 (25870 57704 765632 925000) t nil ((activation (25870 57704 767026 971000) "Activating package" 0) (info (25870 57704 767021 803000) "No Info dir file found" 0) (unblocked (25870 57704 766989 326000) "unblocked by dependency tablist" 0) (queueing-deps (25870 57704 765652 341000) "Queueing Dependencies" 1) (queued (25870 57704 765638 163000) "Package queued" 1)) t))
  elpaca--add-info-path((elpaca pdf-tools "pdf-tools" pdf-tools (activation info unblocked blocked queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/p..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "pdf-tools" :fetcher github :repo "vedang/pdf-tools" :files (:defaults "README" ("build" "Makefile") ("build" "server")) :protocol https :inherit t :depth 1) nil (tablist) (pdf-view-restore) ((emacs "26.3") (tablist "1.0") (let-alist "1.0.4")) 2 (25870 57704 765632 925000) t nil ((activation (25870 57704 767026 971000) "Activating package" 0) (info (25870 57704 767021 803000) "No Info dir file found" 0) (unblocked (25870 57704 766989 326000) "unblocked by dependency tablist" 0) (queueing-deps (25870 57704 765652 341000) "Queueing Dependencies" 1) (queued (25870 57704 765638 163000) "Package queued" 1)) t))
  elpaca--continue-build((elpaca pdf-tools "pdf-tools" pdf-tools (activation info unblocked blocked queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/p..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "pdf-tools" :fetcher github :repo "vedang/pdf-tools" :files (:defaults "README" ("build" "Makefile") ("build" "server")) :protocol https :inherit t :depth 1) nil (tablist) (pdf-view-restore) ((emacs "26.3") (tablist "1.0") (let-alist "1.0.4")) 2 (25870 57704 765632 925000) t nil ((activation (25870 57704 767026 971000) "Activating package" 0) (info (25870 57704 767021 803000) "No Info dir file found" 0) (unblocked (25870 57704 766989 326000) "unblocked by dependency tablist" 0) (queueing-deps (25870 57704 765652 341000) "Queueing Dependencies" 1) (queued (25870 57704 765638 163000) "Package queued" 1)) t) "unblocked by dependency tablist" unblocked)
  elpaca--check-status((elpaca tablist "tablist" tablist (finished activation info queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/t..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "tablist" :fetcher github :repo "emacsorphanage/tablist" :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil nil (pdf-tools) ((emacs "24.3")) 2 (25870 57704 766230 423000) t nil ((activation (25870 57704 766930 729000) "Autoloads cached" 0) (activation (25870 57704 766678 808000) "Caching autoloads" 0) (activation (25870 57704 766656 459000) "Package build dir added to load-path" 0) (activation (25870 57704 766504 483000) "Activating package" 0) (info (25870 57704 766498 337000) "No Info dir file found" 0) (queueing-deps (25870 57704 766450 984000) "No dependencies to queue" 1) (queueing-deps (25870 57704 766258 849000) "Queueing Dependencies" 1) (queued (25870 57704 766235 661000) "Package queued" 1)) t) (elpaca pdf-tools "pdf-tools" pdf-tools (activation info unblocked blocked queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/p..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "pdf-tools" :fetcher github :repo "vedang/pdf-tools" :files (:defaults "README" ("build" "Makefile") ("build" "server")) :protocol https :inherit t :depth 1) nil (tablist) (pdf-view-restore) ((emacs "26.3") (tablist "1.0") (let-alist "1.0.4")) 2 (25870 57704 765632 925000) t nil ((activation (25870 57704 767026 971000) "Activating package" 0) (info (25870 57704 767021 803000) "No Info dir file found" 0) (unblocked (25870 57704 766989 326000) "unblocked by dependency tablist" 0) (queueing-deps (25870 57704 765652 341000) "Queueing Dependencies" 1) (queued (25870 57704 765638 163000) "Package queued" 1)) t))
  elpaca--signal((elpaca tablist "tablist" tablist (finished activation info queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/t..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "tablist" :fetcher github :repo "emacsorphanage/tablist" :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil nil (pdf-tools) ((emacs "24.3")) 2 (25870 57704 766230 423000) t nil ((activation (25870 57704 766930 729000) "Autoloads cached" 0) (activation (25870 57704 766678 808000) "Caching autoloads" 0) (activation (25870 57704 766656 459000) "Package build dir added to load-path" 0) (activation (25870 57704 766504 483000) "Activating package" 0) (info (25870 57704 766498 337000) "No Info dir file found" 0) (queueing-deps (25870 57704 766450 984000) "No dependencies to queue" 1) (queueing-deps (25870 57704 766258 849000) "Queueing Dependencies" 1) (queued (25870 57704 766235 661000) "Package queued" 1)) t) "✓ 0.000 secs" finished)
  elpaca--finalize((elpaca tablist "tablist" tablist (finished activation info queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/t..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "tablist" :fetcher github :repo "emacsorphanage/tablist" :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil nil (pdf-tools) ((emacs "24.3")) 2 (25870 57704 766230 423000) t nil ((activation (25870 57704 766930 729000) "Autoloads cached" 0) (activation (25870 57704 766678 808000) "Caching autoloads" 0) (activation (25870 57704 766656 459000) "Package build dir added to load-path" 0) (activation (25870 57704 766504 483000) "Activating package" 0) (info (25870 57704 766498 337000) "No Info dir file found" 0) (queueing-deps (25870 57704 766450 984000) "No dependencies to queue" 1) (queueing-deps (25870 57704 766258 849000) "Queueing Dependencies" 1) (queued (25870 57704 766235 661000) "Package queued" 1)) t))
  elpaca--continue-build((elpaca tablist "tablist" tablist (finished activation info queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/t..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "tablist" :fetcher github :repo "emacsorphanage/tablist" :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil nil (pdf-tools) ((emacs "24.3")) 2 (25870 57704 766230 423000) t nil ((activation (25870 57704 766930 729000) "Autoloads cached" 0) (activation (25870 57704 766678 808000) "Caching autoloads" 0) (activation (25870 57704 766656 459000) "Package build dir added to load-path" 0) (activation (25870 57704 766504 483000) "Activating package" 0) (info (25870 57704 766498 337000) "No Info dir file found" 0) (queueing-deps (25870 57704 766450 984000) "No dependencies to queue" 1) (queueing-deps (25870 57704 766258 849000) "Queueing Dependencies" 1) (queued (25870 57704 766235 661000) "Package queued" 1)) t))
  elpaca--activate-package((elpaca tablist "tablist" tablist (finished activation info queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/t..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "tablist" :fetcher github :repo "emacsorphanage/tablist" :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil nil (pdf-tools) ((emacs "24.3")) 2 (25870 57704 766230 423000) t nil ((activation (25870 57704 766930 729000) "Autoloads cached" 0) (activation (25870 57704 766678 808000) "Caching autoloads" 0) (activation (25870 57704 766656 459000) "Package build dir added to load-path" 0) (activation (25870 57704 766504 483000) "Activating package" 0) (info (25870 57704 766498 337000) "No Info dir file found" 0) (queueing-deps (25870 57704 766450 984000) "No dependencies to queue" 1) (queueing-deps (25870 57704 766258 849000) "Queueing Dependencies" 1) (queued (25870 57704 766235 661000) "Package queued" 1)) t))
  elpaca--continue-build((elpaca tablist "tablist" tablist (finished activation info queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/t..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "tablist" :fetcher github :repo "emacsorphanage/tablist" :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil nil (pdf-tools) ((emacs "24.3")) 2 (25870 57704 766230 423000) t nil ((activation (25870 57704 766930 729000) "Autoloads cached" 0) (activation (25870 57704 766678 808000) "Caching autoloads" 0) (activation (25870 57704 766656 459000) "Package build dir added to load-path" 0) (activation (25870 57704 766504 483000) "Activating package" 0) (info (25870 57704 766498 337000) "No Info dir file found" 0) (queueing-deps (25870 57704 766450 984000) "No dependencies to queue" 1) (queueing-deps (25870 57704 766258 849000) "Queueing Dependencies" 1) (queued (25870 57704 766235 661000) "Package queued" 1)) t))
  elpaca--add-info-path((elpaca tablist "tablist" tablist (finished activation info queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/t..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "tablist" :fetcher github :repo "emacsorphanage/tablist" :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil nil (pdf-tools) ((emacs "24.3")) 2 (25870 57704 766230 423000) t nil ((activation (25870 57704 766930 729000) "Autoloads cached" 0) (activation (25870 57704 766678 808000) "Caching autoloads" 0) (activation (25870 57704 766656 459000) "Package build dir added to load-path" 0) (activation (25870 57704 766504 483000) "Activating package" 0) (info (25870 57704 766498 337000) "No Info dir file found" 0) (queueing-deps (25870 57704 766450 984000) "No dependencies to queue" 1) (queueing-deps (25870 57704 766258 849000) "Queueing Dependencies" 1) (queued (25870 57704 766235 661000) "Package queued" 1)) t))
  elpaca--continue-build((elpaca tablist "tablist" tablist (finished activation info queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/t..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "tablist" :fetcher github :repo "emacsorphanage/tablist" :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil nil (pdf-tools) ((emacs "24.3")) 2 (25870 57704 766230 423000) t nil ((activation (25870 57704 766930 729000) "Autoloads cached" 0) (activation (25870 57704 766678 808000) "Caching autoloads" 0) (activation (25870 57704 766656 459000) "Package build dir added to load-path" 0) (activation (25870 57704 766504 483000) "Activating package" 0) (info (25870 57704 766498 337000) "No Info dir file found" 0) (queueing-deps (25870 57704 766450 984000) "No dependencies to queue" 1) (queueing-deps (25870 57704 766258 849000) "Queueing Dependencies" 1) (queued (25870 57704 766235 661000) "Package queued" 1)) t) "No dependencies to queue" nil nil 1)
  elpaca--queue-dependencies((elpaca tablist "tablist" tablist (finished activation info queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/t..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "tablist" :fetcher github :repo "emacsorphanage/tablist" :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil nil (pdf-tools) ((emacs "24.3")) 2 (25870 57704 766230 423000) t nil ((activation (25870 57704 766930 729000) "Autoloads cached" 0) (activation (25870 57704 766678 808000) "Caching autoloads" 0) (activation (25870 57704 766656 459000) "Package build dir added to load-path" 0) (activation (25870 57704 766504 483000) "Activating package" 0) (info (25870 57704 766498 337000) "No Info dir file found" 0) (queueing-deps (25870 57704 766450 984000) "No dependencies to queue" 1) (queueing-deps (25870 57704 766258 849000) "Queueing Dependencies" 1) (queued (25870 57704 766235 661000) "Package queued" 1)) t))
  elpaca--continue-build((elpaca tablist "tablist" tablist (finished activation info queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/t..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "tablist" :fetcher github :repo "emacsorphanage/tablist" :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil nil (pdf-tools) ((emacs "24.3")) 2 (25870 57704 766230 423000) t nil ((activation (25870 57704 766930 729000) "Autoloads cached" 0) (activation (25870 57704 766678 808000) "Caching autoloads" 0) (activation (25870 57704 766656 459000) "Package build dir added to load-path" 0) (activation (25870 57704 766504 483000) "Activating package" 0) (info (25870 57704 766498 337000) "No Info dir file found" 0) (queueing-deps (25870 57704 766450 984000) "No dependencies to queue" 1) (queueing-deps (25870 57704 766258 849000) "Queueing Dependencies" 1) (queued (25870 57704 766235 661000) "Package queued" 1)) t))
  elpaca--queue-dependencies((elpaca pdf-tools "pdf-tools" pdf-tools (activation info unblocked blocked queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/p..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "pdf-tools" :fetcher github :repo "vedang/pdf-tools" :files (:defaults "README" ("build" "Makefile") ("build" "server")) :protocol https :inherit t :depth 1) nil (tablist) (pdf-view-restore) ((emacs "26.3") (tablist "1.0") (let-alist "1.0.4")) 2 (25870 57704 765632 925000) t nil ((activation (25870 57704 767026 971000) "Activating package" 0) (info (25870 57704 767021 803000) "No Info dir file found" 0) (unblocked (25870 57704 766989 326000) "unblocked by dependency tablist" 0) (queueing-deps (25870 57704 765652 341000) "Queueing Dependencies" 1) (queued (25870 57704 765638 163000) "Package queued" 1)) t))
  elpaca--continue-build((elpaca pdf-tools "pdf-tools" pdf-tools (activation info unblocked blocked queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/p..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil nil (:package "pdf-tools" :fetcher github :repo "vedang/pdf-tools" :files (:defaults "README" ("build" "Makefile") ("build" "server")) :protocol https :inherit t :depth 1) nil (tablist) (pdf-view-restore) ((emacs "26.3") (tablist "1.0") (let-alist "1.0.4")) 2 (25870 57704 765632 925000) t nil ((activation (25870 57704 767026 971000) "Activating package" 0) (info (25870 57704 767021 803000) "No Info dir file found" 0) (unblocked (25870 57704 766989 326000) "unblocked by dependency tablist" 0) (queueing-deps (25870 57704 765652 341000) "Queueing Dependencies" 1) (queued (25870 57704 765638 163000) "Package queued" 1)) t))
  elpaca--queue-dependencies((elpaca pdf-view-restore "pdf-view-restore" pdf-view-restore (blocked queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/p..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil (elpaca--add-info-path elpaca--activate-package) (:package "pdf-view-restore" :repo "007kevin/pdf-view-restore" :fetcher github :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil (pdf-tools) nil ((pdf-tools "0.90") (emacs "26.0")) 2 (25870 57704 603836 811000) init nil ((blocked (25870 57704 766254 169000) "Blocked by dependencies: (pdf-tools)" 0) (queueing-deps (25870 57704 764393 858000) "Queueing Dependencies" 1) (queued (25870 57704 603844 494000) "Package queued" 1)) t))
  elpaca--continue-build((elpaca pdf-view-restore "pdf-view-restore" pdf-view-restore (blocked queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/p..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil (elpaca--add-info-path elpaca--activate-package) (:package "pdf-view-restore" :repo "007kevin/pdf-view-restore" :fetcher github :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil (pdf-tools) nil ((pdf-tools "0.90") (emacs "26.0")) 2 (25870 57704 603836 811000) init nil ((blocked (25870 57704 766254 169000) "Blocked by dependencies: (pdf-tools)" 0) (queueing-deps (25870 57704 764393 858000) "Queueing Dependencies" 1) (queued (25870 57704 603844 494000) "Package queued" 1)) t))
  elpaca--process((elpaca pdf-view-restore "pdf-view-restore" pdf-view-restore (blocked queueing-deps queued) "/home/icy-thought/.config/emacs/var/elpaca/repos/p..." "/home/icy-thought/.config/emacs/var/elpaca/builds/..." nil nil (elpaca--add-info-path elpaca--activate-package) (:package "pdf-view-restore" :repo "007kevin/pdf-view-restore" :fetcher github :files ("*.el" "*.el.in" "dir" "*.info" "*.texi" "*.texinfo" "doc/dir" "doc/*.info" "doc/*.texi" "doc/*.texinfo" "lisp/*.el" (:exclude ".dir-locals.el" "test.el" "tests.el" "*-test.el" "*-tests.el" "LICENSE" "README*" "*-pkg.el")) :protocol https :inherit t :depth 1) nil (pdf-tools) nil ((pdf-tools "0.90") (emacs "26.0")) 2 (25870 57704 603836 811000) init nil ((blocked (25870 57704 766254 169000) "Blocked by dependencies: (pdf-tools)" 0) (queueing-deps (25870 57704 764393 858000) "Queueing Dependencies" 1) (queued (25870 57704 603844 494000) "Package queued" 1)) t))
  elpaca--process-queue((elpaca-q init 2 21 incomplete ... ... ... ...))
  elpaca-process-queues()
  elpaca-wait()
  load-with-code-conversion("/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/home/icy-thought/Workspace/public/emacs.d/irkalla..." nil t)
  require(init-diagnostics)
  load-with-code-conversion("/home/icy-thought/Workspace/public/emacs.d/irkalla..." "/home/icy-thought/Workspace/public/emacs.d/irkalla..." nil t)
  require(init-coding)
  load-with-code-conversion("/home/icy-thought/.config/emacs/init.el" "/home/icy-thought/.config/emacs/init.el" t t)
  load("/home/icy-thought/.config/emacs/init" noerror nomessage)
  startup--load-user-init-file(#f(compiled-function () #<bytecode 0x9b2f514360b82a0>) #f(compiled-function () #<bytecode -0x1ba2825ef9eb89d9>) t)
  command-line()
  normal-top-level()
```

Emacs config: https://github.com/Icy-Thought/emacs.d/tree/main/irkalla

**Comment by progfolio** [comment]
It looks like pdf-view-restore is pulling in pdf-tools as a dependency.
This shouldn't happen if pdf-tools is a member of `elpaca-ignored-dependencies`.

**Comment by Icy-Thought** [comment]
That seems to have resolved the issue! Appreciate the time and effort you spent on helping me resolve this issue!

**Comment by progfolio** [comment]
No-Name ***@***.***> writes:

> That seems to have resolved the issue! Appreciate the time and 
> effort you spent on helping
> me resolve this issue!

Glad to hear it.
Thanks for taking the time to fill out the issue report.

**Comment by Icy-Thought** [comment]
That's the least I could do! :blush:
